[b-5] Implement sorting

andreyryzhkov opened the pull request:
1) Created sort entity (column, enum)
2) Added QueryConstructor
3) Sorting implementation in DAO service
4) Sorting implementation in Movie service
5) Movie Controller change logic for getting movies data
6) Added tests

andreyryzhkov committed:
[t-1] Create genre cache

andreyryzhkov committed:
[b-5] Implement sorting

1) Created sort entity (column, enum)
2) Added QueryConstructor
3) Sorting implementation in DAO service
4) Sorting implementation in Movie service
5) Movie Controller change logic for getting movies data
6) Added tests

Rooman commented on src/main/java/com/aryzhkov/movieland/dao/MovieDao.java:
```diff
@@ -11,4 +12,8 @@
 
     List<Movie> getByGenre(int id);
+
+    List<Movie> getAll(List<SortColumn> sortColumns);
```
class RequestParams -> MovieRequestParam {
    List<SortColumn>
   String searchText; 
   int page;
}

Rooman commented on src/main/java/com/aryzhkov/movieland/web/controller/MovieController.java:
```diff
@@ -19,7 +19,15 @@
     @GetMapping(produces = MediaType.APPLICATION_JSON_UTF8_VALUE)
-    public List<MovieDto> getAll() {
+    public List<MovieDto> getAll(
+            @RequestParam(value = "rating", required = false) String ratingOrder,
```
@RequestParam(value = "rating", required = false) String ratingOrder
 @RequestParam(value = "rating", required = false) SortOrder ratingOrder

andreyryzhkov committed:
[b-5] Implement sorting

Fixed after review

andreyryzhkov committed:
[b-5] Implement sorting

Fixed after review
1) Created InitBuild for enum transformation
2) Added BuildMovieRequestParam
3) Added tests

andreyryzhkov committed:
[b-5] Implement sorting

Fixed after review
